neurodamus/core/coreneuron_report_config.py

```diff
@@ -0,0 +1,285 @@
+        if spike_path is not None:
+            # Get only the spike file name
+            self.spike_filename = spike_path.rsplit("/", maxsplit=1)[-1]
+
+    @run_only_rank0
+    def dump(self, path: str | Path):
+        path = Path(path).resolve()
+        path.parent.mkdir(parents=True, exist_ok=True)
+        # always override
+        logging.info("Writing coreneuron report config file: %s", path)
+        with open(path, "wb") as f:
+            # number of reports
+            f.write(f"{len(self.reports)}\n".encode())
+            # dump each entry
+            for entry in self.reports.values():
+                entry.dump(f)
+
+            f.write(f"{len(self.pop_offsets)}\n".encode())
+            for k, v in self.pop_offsets.items():
+                if v is not None:
+                    f.write(f"{k} {v}\n".encode())
+                else:
+                    f.write(f"{k}\n".encode())
+
+            if self.spike_filename:
+                f.write(f"{self.spike_filename}\n".encode())
+        logging.info("Done! coreneuron report config was written")
+
+    @classmethod
+    def load(cls, path: str) -> CoreReportConfig:
+        config = cls()
+        with open(path, "rb") as f:
+            # --- Read number of reports ---
+            line = f.readline()
+            if not line:
+                return config  # empty file
+            try:
+                num_reports = int(line.decode().strip())
+            except ValueError as err:
+                raise ValueError(f"Invalid number of reports in file: {line}") from err
+
+            # --- Read each report ---
+            for _ in range(num_reports):
+                entry = CoreReportConfigEntry.load_from_file(f)
+                if entry is None:
+                    raise ValueError("Unexpected EOF while reading reports")
+                config.add_entry(entry)
+
+            # --- Mandatory pop_offsets ---
+            line = f.readline()
+            if not line:
+                raise ValueError("Missing population offsets")
+            pop_count = int(line.decode().strip())
+            pop_offsets = {}
+            for _ in range(pop_count):
+                line = f.readline()
+                if not line:
+                    break
+                parts = line.decode().strip().split()
+                k = parts[0]
+                v = int(parts[1]) if len(parts) > 1 else None
+                pop_offsets[k] = v
+            config.set_pop_offsets(pop_offsets)
+
+            # --- Optional spike filename ---
+            line = f.readline()
+            if line:
+                spike_filename = line.decode().strip()
+                if spike_filename:
+                    config.set_spike_filename(spike_filename)
+
+        return config
+
+    @staticmethod
+    @run_only_rank0
+    def update_file(file_path: str, substitutions: dict[str, dict[str, int]]):
+        """Update the report configuration file with new substitutions."""
```

Comment:
Can you give an example of `substitutions` in the docstring? It is very complex data structure here.